File: reset_db.js
/**
 * Database Reset Script for Edu-Resource LMS
 * This script connects to MongoDB and resets (drops) all collections in the database.
 * Use with caution - all data will be permanently deleted!
 * 
 * Usage: node reset_db.js
 * Force mode: node reset_db.js --force
 */

const path = require('path');
const readline = require('readline');

// Use mongoose from backend's node_modules
const mongoose = require('./backend/node_modules/mongoose');

// Load environment variables from .env file
require('./backend/node_modules/dotenv').config({ path: path.resolve(__dirname, '.env') });

const MONGODB_URI = process.env.MONGODB_URI || 'mongodb://localhost:27017/edu-resource';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function askQuestion(question) {
  return new Promise((resolve) => {
    rl.question(question, (answer) => {
      resolve(answer);
    });
  });
}

async function confirmReset() {
  console.log('\n' + '='.repeat(60));
  console.log('⚠️  WARNING: DATABASE RESET ⚠️');
  console.log('='.repeat(60));
  console.log(`\nThis will DELETE ALL DATA from: ${MONGODB_URI}`);
  console.log('\nAll collections in the database will be dropped.');
  console.log('\n' + '='.repeat(60));

  const response = await askQuestion("\nAre you sure you want to proceed? Type 'YES' to confirm: ");
  return response.trim() === 'YES';
}

async function resetDatabase() {
  try {
    // Connect to MongoDB
    console.log(`\n📡 Connecting to MongoDB: ${MONGODB_URI}`);
    await mongoose.connect(MONGODB_URI);
    console.log('✅ Successfully connected to MongoDB!');

    const db = mongoose.connection.db;

    // Get all collection names
    const collections = await db.listCollections().toArray();

    if (collections.length === 0) {
      console.log('\n📭 Database is already empty. No collections to drop.');
      return true;
    }

    console.log(`\n📋 Found ${collections.length} collection(s):`);
    collections.forEach((coll) => {
      console.log(`   - ${coll.name}`);
    });

    // Drop each collection
    console.log('\n🗑️  Dropping collections...');
    for (const collection of collections) {
      await db.dropCollection(collection.name);
      console.log(`   ✅ Dropped: ${collection.name}`);
    }

    console.log('\n' + '='.repeat(60));
    console.log('✅ DATABASE RESET COMPLETE!');
    console.log('='.repeat(60));
    console.log('\nAll collections have been dropped.');
    console.log('You can now restart your application with a fresh database.');
    console.log('\n💡 Run "npm run create-admin" to create the admin user.');

    return true;
  } catch (error) {
    console.error(`\n❌ Error: ${error.message}`);
    return false;
  } finally {
    await mongoose.connection.close();
    console.log('\n📴 MongoDB connection closed.');
  }
}

async function main() {
  console.log('\n' + '='.repeat(60));
  console.log('     EDU-RESOURCE DATABASE RESET UTILITY');
  console.log('='.repeat(60));

  // Check for --force flag
  const force = process.argv.includes('--force') || process.argv.includes('-f');

  if (!force) {
    const confirmed = await confirmReset();
    if (!confirmed) {
      console.log('\n❌ Operation cancelled.');
      rl.close();
      process.exit(0);
    }
  }

  rl.close();
  const success = await resetDatabase();
  process.exit(success ? 0 : 1);
}

main();

File: reset_db.py
#!/usr/bin/env python3
"""
Database Reset Script for Edu-Resource LMS
This script connects to MongoDB and resets (drops) all collections in the database.
Use with caution - all data will be permanently deleted!
"""

import os
import sys
from pymongo import MongoClient
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Configuration
MONGODB_URI = os.getenv('MONGODB_URI', 'mongodb://localhost:27017/edu-resource')

def confirm_reset():
    """Ask for user confirmation before proceeding."""
    print("\n" + "="*60)
    print("⚠️  WARNING: DATABASE RESET ⚠️")
    print("="*60)
    print(f"\nThis will DELETE ALL DATA from: {MONGODB_URI}")
    print("\nThe following collections will be dropped:")
    print("  - users")
    print("  - courses")
    print("  - assignments")
    print("  - submissions")
    print("  - enrollments")
    print("  - messages")
    print("  - notifications")
    print("  - And any other collections in the database")
    print("\n" + "="*60)
    
    response = input("\nAre you sure you want to proceed? Type 'YES' to confirm: ")
    return response.strip() == 'YES'

def reset_database():
    """Connect to MongoDB and drop all collections."""
    try:
        # Connect to MongoDB
        print(f"\n📡 Connecting to MongoDB: {MONGODB_URI}")
        client = MongoClient(MONGODB_URI)
        
        # Extract database name from URI
        db_name = MONGODB_URI.split('/')[-1].split('?')[0]
        db = client[db_name]
        
        # Test connection
        client.admin.command('ping')
        print("✅ Successfully connected to MongoDB!")
        
        # Get all collection names
        collections = db.list_collection_names()
        
        if not collections:
            print("\n📭 Database is already empty. No collections to drop.")
            return True
        
        print(f"\n📋 Found {len(collections)} collection(s):")
        for coll in collections:
            print(f"   - {coll}")
        
        # Drop each collection
        print("\n🗑️  Dropping collections...")
        for collection_name in collections:
            db[collection_name].drop()
            print(f"   ✅ Dropped: {collection_name}")
        
        print("\n" + "="*60)
        print("✅ DATABASE RESET COMPLETE!")
        print("="*60)
        print(f"\nAll collections in '{db_name}' have been dropped.")
        print("You can now restart your application with a fresh database.")
        
        return True
        
    except Exception as e:
        print(f"\n❌ Error: {e}")
        return False
    finally:
        if 'client' in locals():
            client.close()
            print("\n📴 MongoDB connection closed.")

def reset_with_seed():
    """Reset database and optionally run seed data."""
    reset_database()
    
    response = input("\nWould you like to create an admin user? (y/n): ")
    if response.lower() == 'y':
        print("\n💡 Run 'npm run create-admin' to create the admin user.")

def main():
    """Main entry point."""
    print("\n" + "="*60)
    print("     EDU-RESOURCE DATABASE RESET UTILITY")
    print("="*60)
    
    # Check for --force flag
    force = '--force' in sys.argv or '-f' in sys.argv
    
    if not force:
        if not confirm_reset():
            print("\n❌ Operation cancelled.")
            sys.exit(0)
    
    success = reset_database()
    sys.exit(0 if success else 1)

if __name__ == '__main__':
    main()
